src/index.c

```diff
@@ -3605,6 +3616,7 @@ int git_indexwriter_commit(git_indexwriter *writer)
 		return -1;
 	}
 
+	writer->index->dirty = 0;
```

Comment:
This looks like it belongs to the previous commit, where you introduce the dirty bit?